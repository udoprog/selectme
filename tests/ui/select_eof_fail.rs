pub fn main() {
}

/// Various failures caused by input suddenly ending.
async fn error_eof() {
    selectme::select! { _ };
    selectme::select! { binding };
    selectme::select! { binding = };
    selectme::select! { binding = expr };
    selectme::select! { binding = expr => };
}
