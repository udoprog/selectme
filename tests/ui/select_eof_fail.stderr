error: binding must be followed by a `=`
 --> tests/ui/select_eof_fail.rs:6:25
  |
6 |     selectme::select! { _ };
  |                         ^

error: binding must be followed by a `=`
 --> tests/ui/select_eof_fail.rs:7:25
  |
7 |     selectme::select! { binding };
  |                         ^^^^^^^

error: expected branch expression followed by `=>`
 --> tests/ui/select_eof_fail.rs:8:5
  |
8 |     selectme::select! { binding = };
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `$crate::__support::select` which comes from the expansion of the macro `selectme::select` (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected branch expression followed by `=>`
 --> tests/ui/select_eof_fail.rs:9:35
  |
9 |     selectme::select! { binding = expr };
  |                                   ^^^^

error: expected braced group or expression followed by `,`
  --> tests/ui/select_eof_fail.rs:10:5
   |
10 |     selectme::select! { binding = expr => };
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__support::select` which comes from the expansion of the macro `selectme::select` (in Nightly builds, run with -Z macro-backtrace for more info)
